lib/util/git/index.spec.ts

```diff
@@ -1227,4 +1229,170 @@ describe('util/git/index', { timeout: 10000 }, () => {
       expect(branch).toBe('develop');
     });
   });
+
+  describe('forkMode - normal working', () => {
+    let upstreamBase: any;
+    let upstreamOrigin: any;
+
+    beforeAll(async () => {
+      // create an upstream branch and one extra branch in it
+      upstreamBase = await tmp.dir({ unsafeCleanup: true });
+      const upstream = Git(upstreamBase.path);
+      await upstream.init();
+      const defaultUpsBranch = (
+        await upstream.raw('branch', '--show-current')
+      ).trim();
+      await upstream.addConfig('user.email', 'other@example.com');
+      await upstream.addConfig('user.name', 'Other');
+      await fs.writeFile(upstreamBase.path + '/past_file', 'past');
+      await upstream.addConfig('commit.gpgsign', 'false');
+      await upstream.add(['past_file']);
+      await upstream.commit('past message');
+      await upstream.raw(['checkout', '-B', defaultUpsBranch]);
+      await upstream.checkout(['-b', 'develop', defaultUpsBranch]);
+
+      // clone of upstream on local path
+      upstreamOrigin = await tmp.dir({ unsafeCleanup: true });
+      const upstreamRepo = Git(upstreamOrigin.path);
+      await upstreamRepo.clone(upstreamBase.path, '.', ['--bare']);
+      await upstreamRepo.addConfig('commit.gpgsign', 'false');
+    });
+
+    afterAll(async () => {
+      await upstreamBase?.cleanup();
+      await upstreamOrigin?.cleanup();
+    });
+
+    describe('syncForkWithUpstream()', () => {
+      it('throws unknown error', async () => {
+        tmpDir = await tmp.dir({ unsafeCleanup: true });
```

Comment:
This causes the original `tmpDir` to not be cleaned, use a different variable and call cleanup in afterEach/All.